index.go

```diff
@@ -18,6 +19,10 @@ type queueIndex struct {
 
 // newQueueIndex creates/reads index for a BigQueue
 func newQueueIndex(dataDir string) (*queueIndex, error) {
+	if _, err := os.Stat(dataDir); os.IsNotExist(err) {
```

Comment:
Well, if we are creating the directory, we should move the creation to `NewBigQueue` function instead of here. Index is only supposed to take care of index files.